libraries/botbuilder-core/src/middlewareSet.ts

```diff
@@ -101,12 +101,12 @@ export class MiddlewareSet implements Middleware {
      * ```
      * @param middleware One or more middleware handlers(s) to register.
      */
-    public use(...middleware: (MiddlewareHandler | Middleware)[]): this {
-        middleware.forEach((plugin: any) => {
+    public use(...middlewares: (MiddlewareHandler | Middleware)[]): this {
+        middlewares.forEach((plugin) => {
             if (typeof plugin === 'function') {
                 this.middleware.push(plugin);
             } else if (typeof plugin === 'object' && plugin.onTurn) {
-                this.middleware.push((context: TurnContext, next: Function) => plugin.onTurn(context, next));
+                this.middleware.push((context, next) => plugin.onTurn(context, next));
```

Comment:
I don't know if we'll ever get to this, but I'd like to get your input and formalize the JS SDKs expectation for complex parameters that exist as implementations in the SDK.

We communicate to developers that they have to consume all BF SDK dependencies in lock-step, mostly because that's the extent of our test matrix (packages are tested against the same version)... But also because if a bot consumes two different versions of a given `botbuilder-*` package, it may blow up due to `instanceof` misalignment. And in TypeScript scenarios, the project might not build due to ambiguous types due to multiple versions of a `botbuilder-*` package.

Reply:
And my last but favorite point, interfaces have different semantics between C# and TypeScript at runtime. And Python doesn't have interfaces at all ([`abc`](https://docs.python.org/3/library/abc.html)).